File: java/src/Structural/Proxy/PotOfTeaProxy.java
package Structural.Proxy;


//PotOfTeaProxy.java - the Proxy

public class PotOfTeaProxy implements PotOfTeaInterface {  
   PotOfTea potOfTea;
    
   public PotOfTeaProxy() {}
   
   public void pourTea() {
       potOfTea = new PotOfTea();
       potOfTea.pourTea();
   }
}

File: scala/src/main/scala/Structural/Proxy/PotOfTeaProxy.scala
package Structural.Proxy;


//PotOfTeaProxy - the Proxy

class PotOfTeaProxy extends PotOfTeaInterface {  
   PotOfTea potOfTea;
    
   PotOfTeaProxy() {}
   
   String pourTea() {
       potOfTea = new PotOfTea();
       return potOfTea.pourTea();
   }
}
